packages/libs/error-stack/src/report.rs

```diff
@@ -193,18 +193,16 @@ use crate::{
 #[must_use]
 #[repr(transparent)]
 pub struct Report<C> {
-    pub(super) frames: Vec<Frame>,
+    // reason: This implies a memory footprint equal to a single pointer size instead of three
+    //         pointer sizes. `Report` is unlikely to be used in the happy path, but it's almost
+    //         always used in `Result::Err`, so for small `Result::Ok` variants, this would
+    //         increase the size of the `Result` to 24 bytes (on 64 bit) instead of 8 bytes.
```

Comment:
Can we be a little more specific about _when_ this decreases the memory footprint? I think when talking about the memory footprint of a struct it can be unclear if you're talking about the direct contiguous footprint or also the footprint of the other objects on the heap owned by the struct. I appreciate here we're talking about the former, but I'm not sure the comment quite explains why this is beneficial.

Is it that the `Result` will always take up 24 bytes even there isn't a report and therefore there is no Vec? Boxing it avoids the additional space being allocated?

Reply:
(Assuming 64 bit pointer width) A `Vec` will *always* have 24 bytes in size, it contains a pointer, a capacity, and a size. Boxing it will always make it 8 bytes in size, a test for this was added to ensure this in the future.

Reply:
> Is it that the `Result` will always take up those 24 bytes _even if there isn't a report_ and therefore there is no Vec? Boxing it avoids the additional space being allocated?

Yes, even `Result::<(), Report<_>>::Ok(())` will take 24 bytes. Regardless of `Result::<_, Box<_>>` or  `Result::<_, Box<Vec<_>>>` there will be no allocation, but the size of `Box<Vec<_>>` is smaller than `Vec<_>`: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=87ea9db3cf8f19a977116cbd25f941ee).

Reply:
Should we update the comment to explain that more clearly? Just to clarify that boxing it is reducing the memory footprint _on_ the happy path. Because "`Report` is unlikely to be used in the happy path, but"... confuses things a bit in my eyes